spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -455,141 +456,148 @@ def add_electrodes(recording: se.RecordingExtractor, nwbfile=None,
+
+                if not any([x.get('name', '') == 'group_name' for x in metadata['Ecephys']['Electrodes']]):
+                    group_id = recording.get_channel_groups(channel_ids=channel_id)[0]
+                    if group_id in range(len(nwbfile.electrode_groups)):
+                        group_name = list(nwbfile.electrode_groups.keys())[group_id]
+                        electrode_kwargs.update(
+                            dict(
+                                group=nwbfile.electrode_groups[group_name],
+                                group_name=group_name
+                            )
+                        )
                     else:
-                        print("Warning: No metadata was passed specifying the electrode group for"
-                              f" electrode {channel_id}, and the internal recording channel group was"
-                              f" assigned a value ({grp_id}) outside the indices of the electrode"
-                              " groups in the nwbfile! Electrode will not be added.")
+                        warnings.warn("No metadata was passed specifying the electrode group for "
+                                      f"electrode {channel_id}, and the internal recording channel group was "
+                                      f"assigned a value ({group_id}) outside the indices of the electrode "
+                                      "groups in the nwbfile! Electrode will not be added.")
                         continue
 
                 nwbfile.add_electrode(**electrode_kwargs)
 
-        # Add any additional custom columns from data specified via channel properties
         # property 'gain' should not be in the NWB electrodes_table
-        # property 'location' of RX channels corresponds to rel_x and rel_ y of NWB electrodes
-        # and rel_x, rel_y have already been added to this point
+        # property 'brain_area' of RX channels corresponds to 'location' of NWB electrodes
         channel_prop_names = set(recording.get_shared_channel_property_names()) - set(nwbfile.electrodes.colnames) \
-                             - set(['gain', 'location'])
+            - set(['gain', 'location'])
         for channel_prop_name in channel_prop_names:
-            for channel_id in channel_ids:
+            for channel_id in recording.get_channel_ids():
                 val = recording.get_channel_property(channel_id, channel_prop_name)
                 descr = 'no description'
-                # property 'brain_area' of RX channels corresponds to 'location' of NWB electrodes
                 if channel_prop_name == 'brain_area':
-                    if 'location' in nwbfile.electrodes.colnames:
-                        continue
-                    else:
-                        channel_prop_name = 'location'
-                        descr = 'brain area location'
+                    channel_prop_name = 'location'
+                    descr = 'brain area location'
```

Comment:
So currently this does not allow to save properties that are not shared among all channels. Should we keep it that way? Otherwise, we could add `np.nans` for columns that are missing that property

Reply:
If we decide not to keep things that way, it would probably require a more holistic refactor of the way we treat properties in the Recording and Sorting Extractors. For now let's table that. Upon usage of `write_recording` and `write_sorting`, I currently assume such data has been completed by the user (i.e., if there is missing data for any channel or unit, it has been set to some default value by the user upon establishing the properties for that extractor object).

The current refactor shouldn't change the way it was previously handled, which again assumes complete shared data for the internal properties.